Change the color of airline

alexanderjeurissen opened the pull request:
This fixes #23 

The default airline color used for insert mode doesn't really fit the overall look of gotham. Also the white on yellow makes it very hard to read that the active mode is insert.

This changes the color used for insert mode to magenta. See the screenshots below for (before / after)

![screen shot 2016-02-26 at 21 22 41](https://cloud.githubusercontent.com/assets/1220084/13364488/591d23d2-dccf-11e5-8789-69a90ef7113c.png)

![screen shot 2016-02-26 at 21 24 12](https://cloud.githubusercontent.com/assets/1220084/13364487/591a3bae-dccf-11e5-9670-a938d7722554.png)

alexanderjeurissen commented:
I took another look and see your using a different color in the light line colorscheme. Don't merge this yet It's probably better to use the same colors in both colorschemes.

I'll have a look at the differences.

alexanderjeurissen commented:
Also replace mode looks real weird in airline lol:

![screen shot 2016-02-26 at 21 29 19](https://cloud.githubusercontent.com/assets/1220084/13364612/139b3e92-dcd0-11e5-9129-b5a47bc66866.png)

alexanderjeurissen commented:
Ok so I changed several colors in the airline theme so they match that of lightline. See the screenshots below for the before / after

**before:**
![screen shot 2016-02-26 at 21 49 24](https://cloud.githubusercontent.com/assets/1220084/13365046/0862920c-dcd3-11e5-839c-6d7128e0df70.png)
![screen shot 2016-02-26 at 21 49 32](https://cloud.githubusercontent.com/assets/1220084/13365045/08620e68-dcd3-11e5-8935-0922232619a0.png)
![screen shot 2016-02-26 at 21 49 42](https://cloud.githubusercontent.com/assets/1220084/13365048/0863b876-dcd3-11e5-8c9f-c7c868d7a2ae.png)
![screen shot 2016-02-26 at 21 49 52](https://cloud.githubusercontent.com/assets/1220084/13365047/0863c62c-dcd3-11e5-8de1-f5687f62d044.png)

**after:**
![screen shot 2016-02-26 at 21 46 24](https://cloud.githubusercontent.com/assets/1220084/13365053/16f64192-dcd3-11e5-871d-7af1a537a0c0.png)
![screen shot 2016-02-26 at 21 46 33](https://cloud.githubusercontent.com/assets/1220084/13365054/170e5bba-dcd3-11e5-8620-ecfca025a5cd.png)
![screen shot 2016-02-26 at 21 46 56](https://cloud.githubusercontent.com/assets/1220084/13365056/17228f7c-dcd3-11e5-8ebe-006eaffbdec8.png)
![screen shot 2016-02-26 at 21 46 46](https://cloud.githubusercontent.com/assets/1220084/13365057/172477ba-dcd3-11e5-9146-133b9227aa0e.png)

alexanderjeurissen commented:
I'll commit the changes above in a sec.

whatyouhide commented:
@alexanderjeurissen hey, thanks for the contribution (thorough as always :smiley:). A couple of thoughts:

* I'm still not sure about the changes you did to the normal mode. It's hard to read the text now, but I'm not 100% sure it's bad because the theme is very dark; reading the filename in the middle of the modeline is still harder to read, so it may make sense. Let me think about it for a bit.
* I'm inclined to go :-1: on the change to insert mode as I like the idea of having insert mode stand out, since you shouldn't be in insert mode for long in vim and when you are, I like the idea of being reminded that you are :). Wdyt? (btw, changing the text from light to dark still is a great improvement)
* I love the new style for visual mode, let's keep it :+1:
* I like the new style for replace mode, let's keep it :+1:

alexanderjeurissen commented:
>I'm still not sure about the changes you did to the normal mode. It's hard to read the text now, but I'm not 100% sure it's bad because the theme is very dark; reading the filename in the middle of the modeline is still harder to read, so it may make sense. Let me think about it for a bit.

I've tried it for some days now and I find the text quite readable I could lighten it a bit though as I've currently used `base3` instead of `base4`.

>I'm inclined to go :-1: on the change to insert mode as I like the idea of having insert mode stand out, since you shouldn't be in insert mode for long in vim and when you are, I like the idea of being reminded that you are :). Wdyt? (btw, changing the text from light to dark still is a great improvement)

I agree with the fact that dark text is more readable on the yellow. However the point about not being in insert mode, the same can be said about visual mode because most actions where visual mode is used can be done using text-objects or using relative line numbers. That being said the other colors feel quite nice because they are also used in the colorscheme, however the yellow isn't used a lot in code highlighting and that makes it feel out of place since the yellow doesn't quite match the dark and matt theme of gotham.

One option could be to add a feature toggle that could be set in people their vimrc to make insert and visual mode stand out more. Other colorschemes have similar configuration variables where they can increase or decrease the contrast of the colorscheme for example. (see solarized and base16)

whatyouhide commented:
I am :-1: against adding options for this. Let's:

* revert the "branch" and "file path" parts to the previous setting for both visual and replace, I don't like the purple-ish new setting (and I like that visual/replace/insert are the same on those parts).
* bump the text for normal mode from `base3` to `base4` as it's too dark for my tastes the way it is now
* revert insert to yellow background with dark foreground, as I still think it's nice that it stands out so much as you really shouldn't be in insert mode for long.

Toughts?

alexanderjeurissen commented:
Sorry for the late response.I'll address your suggestions but before I do May I ask why your against adding options for this ? options for increased contrast or diffing is quite usual in vim colorscheme land. As it allows for experimental features and/or features that only a minority of people use or want.

now to address your suggestions:

the purpelish font of the filename is because there were modifications in the file. They don't show up if you have no modifications and this color is currently being used already in those circumstances (hence is not a change I made)

I'l bump the text color for normal mode :+1: 

I think the new insert mode stands out quite a bit because the green is very bright compared with the rest of the editor colors. I agree that you shouldn't be in insert mode a lot when writing code but I also happen to write blogs and documentation in vim using markdown and in those circumstances I'm in insert mode quite often. The yellow then kind of removes the nice dark atmosphere that makes the gotham theme such a pleasure to use.

whatyouhide commented:
You can try to add an option for increased contrast (which would mean yellow insert mode) and we can see how it looks.

alexanderjeurissen commented:
I'll work on that somewhere in the weekend.

whatyouhide commented:
Ping?

alexanderjeurissen commented:
@whatyouhide sorry for the delay, I'm currently using another colorscheme so recently haven't work on this. I can tackle this next weekend.

whatyouhide commented:
Ping @alexanderjeurissen? :)

alexanderjeurissen commented:
Sorry for the delay was really busy. I just updated the PR and will commit it shortly. This is the final result:
![screen shot 2017-02-04 at 15 07 41](https://cloud.githubusercontent.com/assets/1220084/22619068/3d52ca90-eaec-11e6-8f20-ebfd1bdcaccb.png)
![screen shot 2017-02-04 at 15 07 35](https://cloud.githubusercontent.com/assets/1220084/22619069/3d5410b2-eaec-11e6-8f48-c198eae9300a.png)
![screen shot 2017-02-04 at 15 07 27](https://cloud.githubusercontent.com/assets/1220084/22619071/3d560f48-eaec-11e6-90d4-5dd1d69be04a.png)
![screen shot 2017-02-04 at 15 07 06](https://cloud.githubusercontent.com/assets/1220084/22619073/3d5921ce-eaec-11e6-974f-1543ef6eece7.png)
![screen shot 2017-02-04 at 15 04 25](https://cloud.githubusercontent.com/assets/1220084/22619070/3d54a126-eaec-11e6-8801-4cbd0add08fb.png)
![screen shot 2017-02-04 at 15 04 11](https://cloud.githubusercontent.com/assets/1220084/22619072/3d584a9c-eaec-11e6-832a-c304dcb9201d.png)
![screen shot 2017-02-04 at 15 04 04](https://cloud.githubusercontent.com/assets/1220084/22619074/3d689c58-eaec-11e6-9666-27ecb3a34d32.png)


I added an option called: `g:gotham_airline_contrast_mode` which when set to 1 changes the color of insert mode to this:
![screen shot 2017-02-04 at 15 11 11](https://cloud.githubusercontent.com/assets/1220084/22619080/5bacc112-eaec-11e6-8c70-04d2d3c82b29.png)


As you can see I increased the visibility of the mode text by changing `base3` to `base2` I'm quite pleased with the result.

alexanderjeurissen commented:
I just rebased on the latest master branch.

alexanderjeurissen commented:
I just checked and vimscript `get(` always returns true if the variable is not set. This might be the correct behaviour if we want to default to the yellow insert, otherwise we need to include an explicit statement like:
```viml
let g:gotham_airline_contrast_mode = 0
```

in the colorscheme file, and ask people to override it. WDYT ?

whatyouhide commented:
Let's default to contrasty insert, and let users override this if they want to.

alexanderjeurissen commented:
@whatyouhide which of the two values do you refer to with 'contrasty insert' you mean default to yellow insert ?

if that is the case then I have to change the readme diff in this PR.

whatyouhide commented:
@alexanderjeurissen yes, sorry for not being clear, I meant let's default to yellow insert and make non-yellow insert configurable. :)

alexanderjeurissen commented:
@whatyouhide updated the readme, and changed the name of the config variable to: `g:gotham_airline_emphasised_insert`

whatyouhide commented:
Thanks a lot @alexanderjeurissen! 💟
